staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go

```diff
@@ -2,111 +2,334 @@ package context
 
 import (
 	gocontext "context"
+	"fmt"
 	"sync"
 	"time"
 
 	"k8s.io/klog/v2"
 
+	"github.com/kubeedge/beehive/pkg/common"
+	"github.com/kubeedge/beehive/pkg/core/channel"
 	"github.com/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/beehive/pkg/core/socket"
+	"github.com/kubeedge/beehive/pkg/core/socket/config"
 )
 
-// define channel type
-const (
-	MsgCtxTypeChannel = "channel"
-)
+// BeehiveContext is global context: only use for local cache to dispatch message
+type BeehiveContext struct {
```

Comment:
```suggestion
type beehiveContext struct {
```
?

Reply:
this structure is also used by `staging/src/github.com/kubeedge/beehive/pkg/core/core.go`, need to be exported.